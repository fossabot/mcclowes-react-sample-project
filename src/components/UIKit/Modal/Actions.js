import React from "react";
import styled from "styled-components";

const ActionsWrapper = styled.div`
  display: flex;
  justify-content: flex-end;
`;

const Actions = props => {
  const { children, ...rest } = props;

  return <ActionsWrapper {...rest}>{children}</ActionsWrapper>;
};

export default Actions;
