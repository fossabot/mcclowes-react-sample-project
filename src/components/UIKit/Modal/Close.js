import React from "react";
import styled from "styled-components";
import Icon, { Icons } from "../Icon";

const CloseWrapper = styled("div")`
  position: absolute;
  right: 10px;
  top: 10px;
  outline: none;
  z-index: 5;

  &,
  * {
    cursor: pointer;
  }
`;

const Close = props => {
  const { icon, onClick, ...rest } = props;

  return (
    <CloseWrapper {...rest}>
      {icon ? (
        icon
      ) : (
        <button onClick={onClick}>
          <Icon path={Icons.CLOSE} />
        </button>
      )}
    </CloseWrapper>
  );
};

export default Close;
