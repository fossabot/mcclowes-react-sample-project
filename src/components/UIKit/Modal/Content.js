import React from "react";
import styled from "styled-components";

const ContentWrapper = styled.div`
  min-height: 100%;
`;

const Content = props => {
  const { children, ...rest } = props;

  return <ContentWrapper {...rest}>{children}</ContentWrapper>;
};

export default Content;
