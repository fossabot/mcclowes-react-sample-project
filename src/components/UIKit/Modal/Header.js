import React from "react";
import styled from "styled-components";

const HeaderWrapper = styled.div``;

const Header = props => {
  const { children, ...rest } = props;

  return <HeaderWrapper {...rest}>{children}</HeaderWrapper>;
};

export default Header;
