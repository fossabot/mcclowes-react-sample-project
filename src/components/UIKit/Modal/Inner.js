import React from "react";
import styled from "styled-components";

const InnerWrapper = styled.div`
  background-color: white;
  margin: 0 !important;
  max-height: 100%;
  max-height: 90vh;
  max-width: 100%;
  min-width: 680px;
  overflow: auto;

  ${props =>
    props["data-modal-size"] === "modal-mini" &&
    `
    min-width: 318px;
    max-width: 318px;
  `};

  ${props =>
    props["data-modal-size"] === "modal-tiny" &&
    `
    max-width: 500px;
    min-width: 500px;
  `};

  ${props =>
    props["data-modal-size"] === "modal-small" &&
    `
    max-width: 680px;
  `};

  ${props =>
    props["data-modal-size"] === "modal-medium" &&
    `
    max-width: 920px;
  `};
`;

const Inner = props => {
  const { children, size, ...rest } = props;

  return (
    <InnerWrapper data-modal-size={size && `modal-${size}`} {...rest}>
      {children}
    </InnerWrapper>
  );
};

export default Inner;
