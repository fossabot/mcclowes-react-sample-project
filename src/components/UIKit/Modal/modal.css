.modal-overlay {
  align-items: center;
  background-color: rgba(50, 50, 50, 0.75);
  display: flex;
  justify-content: center;
  transition: opacity 200ms ease-in-out;
  z-index: 1000;
  padding: 1em;
}

.modal-wrapper {
  background: none;
  border: none;
  border-radius: 5px;
  bottom: auto;
  left: auto;
  max-height: 90vh;
  max-width: 90vw;
  overflow: hidden;
  padding: 0;
  position: relative;
  right: auto;
  top: auto;
  display: flex;
  align-items: center;
}
