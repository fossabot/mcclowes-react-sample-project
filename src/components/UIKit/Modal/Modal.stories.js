import React from "react";
import Modal from "./Modal";
import Button from "../Button";

export default { title: "Modal" };

export const simple = () => {
  return (
    <Modal open>
      <Modal.Inner>Hello world</Modal.Inner>
    </Modal>
  );
};

export const structured = () => {
  return (
    <Modal open closeIcon doClose={() => {}}>
      <Modal.Inner>
        <Modal.Header>Header</Modal.Header>

        <Modal.Content>
          Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
          Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis
          vestibulum. Nulla vitae elit libero, a pharetra augue. Duis mollis,
          est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio
          sem nec elit. Cras mattis consectetur purus sit amet fermentum.
          Vestibulum id ligula porta felis euismod semper. Lorem ipsum dolor sit
          amet, consectetur adipiscing elit.
        </Modal.Content>

        <Modal.Actions>
          <Button>Close</Button>
          <Button>Submit</Button>
        </Modal.Actions>
      </Modal.Inner>
    </Modal>
  );
};

export const controlled = () => {
  return (
    <Modal open>
      <Modal.Inner>
        <Modal.Header>Header</Modal.Header>

        <Modal.Content>
          Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
          Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis
          vestibulum. Nulla vitae elit libero, a pharetra augue. Duis mollis,
          est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio
          sem nec elit. Cras mattis consectetur purus sit amet fermentum.
          Vestibulum id ligula porta felis euismod semper. Lorem ipsum dolor sit
          amet, consectetur adipiscing elit.
        </Modal.Content>

        <Modal.Actions>Hello world</Modal.Actions>
      </Modal.Inner>
    </Modal>
  );
};
