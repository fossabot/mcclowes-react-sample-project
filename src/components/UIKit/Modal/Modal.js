import PropTypes from "prop-types";
import React from "react";
import ReactModal from "react-modal";

import Actions from "./Actions";
import Close from "./Close";
import Content from "./Content";
import Header from "./Header";
import Inner from "./Inner";

const defaultStyles = {
  overlay: {
    alignItems: "center",
    backgroundColor: "rgba(50,50,50,.75)",
    display: "flex",
    justifyContent: "center",
    transition: "opacity 200ms ease-in-out",
    zIndex: 1000 // must be equal to semantics modal
  },
  content: {
    background: "none",
    border: "none",
    borderRadius: "3px",
    bottom: "auto",
    left: "auto",
    maxHeight: "90vh",
    maxWidth: "90vw",
    overflow: "hidden",
    padding: "0",
    position: "relative",
    right: "auto",
    top: "auto",
    display: "flex",
    alignItems: "center"
  }
};

if (process.env.NODE_ENV !== "test") ReactModal.setAppElement("#root");

const Modal = props => {
  const { children, doClose, open, closeIcon, modalStyles, trigger } = props;

  //const { overlay, content } = modalStyles;

  const style = {
    overlay: {
      ...defaultStyles.overlay
      //...overlay
    },
    content: {
      ...defaultStyles.content
      //...content
    }
  };

  return (
    <>
      {trigger && trigger}

      <div onClick={e => e.stopPropagation()}>
        <ReactModal
          isOpen={open}
          closeTimeoutMS={400}
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          onRequestClose={doClose}
          style={style}
        >
          {closeIcon && doClose && <Close onClick={doClose} />}

          {children}
        </ReactModal>
      </div>
    </>
  );
};

Modal.Actions = Actions;
Modal.Close = Close;
Modal.Content = Content;
Modal.Header = Header;
Modal.Inner = Inner;

Modal.propTypes = {
  children: PropTypes.node,
  doClose: PropTypes.func,
  trigger: PropTypes.any
};

export default Modal;
