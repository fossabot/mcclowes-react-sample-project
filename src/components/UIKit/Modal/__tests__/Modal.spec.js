import Modal from "../Modal";
import React from "react";
import ReactModal from "react-modal";
import { shallow } from "enzyme";

const ContentMock = () => {
  return <div>This is in a portal</div>;
};

describe.skip("Modal", () => {
  const shallowComponent = (props = {}) => {
    return shallow(
      <Modal isOpen={true} doClose={jest.fn()} {...props}>
        <ContentMock />
      </Modal>
    );
  };

  const shallowAndFind = props => {
    return shallowComponent(props).find(ReactModal);
  };

  it("renders as expected", () => {
    const wrapper = shallowComponent();

    expect(wrapper).toMatchSnapshot();
  });
});
