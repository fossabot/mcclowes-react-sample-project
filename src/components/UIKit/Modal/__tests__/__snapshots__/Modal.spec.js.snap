// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders as expected 1`] = `
<Fragment>
  <div
    onClick={[Function]}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={400}
      isOpen={true}
      onRequestClose={[MockFunction]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "alignItems": "center",
            "background": "none",
            "border": "none",
            "borderRadius": "3px",
            "bottom": "auto",
            "display": "flex",
            "left": "auto",
            "maxHeight": "90vh",
            "maxWidth": "90vw",
            "overflow": "hidden",
            "padding": "0",
            "position": "relative",
            "right": "auto",
            "top": "auto",
          },
          "overlay": Object {
            "alignItems": "center",
            "backgroundColor": "rgba(50,50,50,.75)",
            "display": "flex",
            "justifyContent": "center",
            "transition": "opacity 200ms ease-in-out",
            "zIndex": 1000,
          },
        }
      }
    >
      <ContentMock />
    </Modal>
  </div>
</Fragment>
`;
